Dockerfile

```diff
@@ -119,7 +134,9 @@ RUN groupadd -g 1000 zeebe && \
     chmod -R 0775 ${ZB_HOME}
 
 COPY --from=init --chown=1000:0 /zeebe/tini ${ZB_HOME}/bin/
+COPY --from=init --chown=1000:0 /zeebe/gosu ${ZB_HOME}/bin/
 COPY --from=init --chown=1000:0 /zeebe/startup.sh /usr/local/bin/startup.sh
 COPY --from=dist --chown=1000:0 /zeebe/camunda-zeebe ${ZB_HOME}
 
-ENTRYPOINT ["tini", "--", "/usr/local/bin/startup.sh"]
+
+ENTRYPOINT ["gosu", "zeebe:zeebe", "tini", "--", "/usr/local/bin/startup.sh"]
```

Comment:
Hey @megglos 

I relied on
https://github.com/krallin/tini/issues/74
https://stackoverflow.com/questions/36781372/docker-using-gosu-vs-user

Reply:
Hey @mvawork,
thanks for providing these references!

For allowing to get a root shell on a running container you can actually still specify root via the [--user flag](https://docs.docker.com/engine/reference/commandline/exec/).

Where I now see the use case for gosu though is actually making this change non-breaking, as currently the rolling update tests fail because of permission issues.
We could keep root as the default user in the Dockerfile and change ownership of potentially existing data in the `data` and logs` volumes in the `startup.sh` script and then drop from root to the `zeebe` user to run the zeebe process.

wdyt? could you make these changes?

We would also need to follow-up on this for the helm chart as it overrides the startup script, I could take if from there though once we have a non-breaking solution in place.

Reply:
I will actually verify the effect on helm deployments, it could be that the permission issue isn't happening there in the way it happens on plain docker.

Reply:
Hey @megglos, I wanted to let you know that in general, kubernetes applications have the `fsGroup` securityContext option, such that if the current user is different from the UID and GID of the volumes,  the pod will change the volumes of the volumes before mounting. Our helm charts do provide options for providing this security context, and I have instructed some users how to set these options.